pkg/cmd/roachtest/tests/rapid_restart.go

```diff
@@ -51,9 +51,14 @@ func runRapidRestart(ctx context.Context, t test.Test, c cluster.Cluster) {
 			go func() {
 				err := c.RunE(ctx, nodes,
 					`mkdir -p {log-dir} && ./cockroach start-single-node --insecure --store={store-dir} `+
-						`--log-dir={log-dir} --cache=10% --max-sql-memory=10% `+
-						`--listen-addr=:{pgport:1} --http-port=$[{pgport:1}+1] `+
+						`--log='file-defaults: dir: {log-dir}' --cache=10% --max-sql-memory=10% `+
```

Comment:
I don't think this `--log` syntax is correct after all. Have you actually tried it?

Reply:
I did. Seemed error-free. what's wrong with the syntax? is it because it is supposed to be `--log='file-defaults: {dir: log-dir}` or is it supposed to be the original value?

Reply:
yes, I'm getting the following error:
```
% ./cockroach start-single-node --insecure --log='file-defaults: dir: foo'
E210930 18:59:53.860309 1 1@cli/clierror/check.go:35  [-] 1  ERROR: yaml: mapping values are not allowed in this context
ERROR: yaml: mapping values are not allowed in this context
```

Also, I think that `{dir: log-dir}` won't work? The syntax "`{log-dir}`" is a special string that is interpreted by roachprod to inject the logging directory, but I don't know that it recognizes braces in other locations.

Reply:
ah yes I see. hmm. in that case maybe it is best to stick with `--log-dir`. It seems like the deprecated http flag was the true problem anyway.

Reply:
sounds good!